### ISSUE TITLE
Support extended character sets in install.conf.yaml 🐈

### ISSUE THREAD

**Issue by jkrauska** [anishathalye/dotbot#205]
I wanted to include a file named 🐈 in my dotfiles.

```
$ ./install
Could not read config file:
  unacceptable character #xd83d: special characters are not allowed
    in "install.conf.yaml", position 162
```

Please consider supporting extended character sets?

**Comment by jkrauska** [comment]
Seems to be a pyyaml bug.

https://stackoverflow.com/questions/44875403/loading-special-characters-with-pyyaml

**Comment by anishathalye** [comment]
It's kinda fixed in PyYAML 5.1.2. Dotbot 2c8a0431edfd826c8caa4afbfa22ca408e745db6 switches to that version, so emoji kinda works.

The support depends on `sys.maxunicode`, so for me, it was working with Python 3, but not with Python 2. I opened an issue in PyYAML for this: https://github.com/yaml/pyyaml/issues/350

Until this works in both Python 2 and Python 3, support for emoji won't be added to Dotbot's spec (/ won't add tests for it), but as of the latest Dotbot commit, emojis etc do seem to work with newer versions of Python.

**Comment by anishathalye** [comment]
I submitted a PR to PyYAML to fix the issue with Python 2 (UCS-2) support: https://github.com/yaml/pyyaml/pull/351

**Comment by anishathalye** [comment]
That PR has been merged into PyYAML and released in 5.3.

Dotbot now [supports](f8100dcf4fd640d6a81d82bc1dd8c75f7f417142) emoji!

**Comment by anishathalye** [comment]
I bumped the PyYAML version to 5.3 in 5d83f9e797b1950199e127a8196803f5e33e0916. The emoji filenames should probably work for you now.

I'm not going to bother with adding the test because it was annoying to get it fully working cross-platform (got it working with cpython 2/3, and pypy2, but was having trouble with pypy3), and it didn't really seem worth it.

The emoji filename stuff will probably work for you unless you're using pypy3. If anyone is particularly enthusiastic about trying to get this to work on pypy3, feel free to reopen / submit a PR.
